### PR TITLE
add 111raw

### LINKED ISSUE #284: 111raw
### Source name

111raw

### Source link

https://111raw.com/

### Source language

Japanese

### Other details

18+/NSFW = yes

### Acknowledgements

- [X] I have checked that the extension does not already exist by searching the [GitHub repository](https://github.com/keiyoushi/extensions-source/) and verified it does not appear in the code base.
- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a meaningful title with the source name.
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by Secozzi**
Closes #284

Checklist:

- [ ] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [ ] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [x] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [x] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [ ] Have not changed source names
- [ ] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio
- [x] Have removed `web_hi_res_512.png` when adding a new extension

### CODE REVIEW

File: src/ja/oneoneoneraw/src/eu/kanade/tachiyomi/extension/ja/oneoneoneraw/OneOneOneRaw.kt [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+    private fun ListDto.EntryDto.toSManga(): SManga = SManga.create().apply {
+        title = name
+        url = "/manga/$slug-$mangaId"
+        thumbnail_url = image
+        description = buildString {
+            this@toSManga.description?.let {
+                append(it)
+                altName?.let { append("\n\n") }
+            }
+            altName?.let { append("Alternative name(s): $it") }
```
Comment (beer-psi) [inline src/ja/oneoneoneraw/src/eu/kanade/tachiyomi/extension/ja/oneoneoneraw/OneOneOneRaw.kt:180]:
```suggestion
            }
            altName?.let {
                append("\n\n")
                append("Alternative name(s): ")
                append(it)
            }
```
Using string interpolation will create a StringBuilder internally.

File: src/ja/oneoneoneraw/src/eu/kanade/tachiyomi/extension/ja/oneoneoneraw/OneOneOneRaw.kt [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+                addEncodedQueryParameter("name", query)
+            } else {
+                addQueryParameter("order_by", filters.firstInstanceOrNull<OrderByFilter>()?.selectedValue())
+            }
+            if (page > 1) {
+                addQueryParameter("page", page.toString())
+            }
+        }
+
+        return GET(url.build(), apiHeaders)
```
Comment (beer-psi) [inline src/ja/oneoneoneraw/src/eu/kanade/tachiyomi/extension/ja/oneoneoneraw/OneOneOneRaw.kt:85]:
```suggestion
        }.build()

        return GET(url, apiHeaders)
```

**Review by beer-psi — CHANGES_REQUESTED** [review]

File: src/ja/oneoneoneraw/src/eu/kanade/tachiyomi/extension/ja/oneoneoneraw/OneOneOneRaw.kt [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+
+    override val baseUrl = "https://111raw.com"
+
+    override val supportsLatest = true
+
+    override val client = network.cloudflareClient.newBuilder()
+        .rateLimit(2)
+        .build()
+
+    private val apiUrl = "https://api.rawz.org/api/manga".toHttpUrl()
```
Comment (AwkwardPeak7) [inline src/ja/oneoneoneraw/src/eu/kanade/tachiyomi/extension/ja/oneoneoneraw/OneOneOneRaw.kt:41]: Is it mirror/alt site to rawz perhaps?
https://github.com/keiyoushi/extensions-source/blob/5a1721eb8760d6137b62709c4e94e8a2a6dffd54/src/ja/rawz/src/eu/kanade/tachiyomi/extension/ja/rawz/RawZ.kt#L27

Comment (Secozzi) [reply src/ja/oneoneoneraw/src/eu/kanade/tachiyomi/extension/ja/oneoneoneraw/OneOneOneRaw.kt:41]: Yup, appears to be an alt/scraper, so I don't see any point in having both as an extension. Could probably close #284.

**Pull request closed without merging**